notebooks/bedrock/Financial Multilingual Embeddings.ipynb

```diff
@@ -0,0 +1,545 @@
+    "\n",
+    "df ['text'] = df['text'].apply(remove_duplicates)\n",
+    "\n",
+    "# Select languages\n",
+    "languages = ['English', 'Spanish', 'Danish']\n",
+    "top_80_df = df.loc[df['lang'].isin(languages)]\n",
+    "\n",
+    "# Pick the top 80 longest articles\n",
+    "top_80_df['text_length'] = top_80_df['text'].str.len()\n",
+    "top_80_df.sort_values(by=['text_length'], ascending=False, inplace=True)\n",
+    "top_80_df = top_80_df[:80]\n",
+    "\n",
+    "# Language distribution\n",
+    "top_80_df['lang'].value_counts()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "09fb0837-91c2-4a5d-8f9a-9772690036f7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's look at the first record\n",
+    "top_80_df['text'].iloc[0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9aa02a01",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up translator. You can also commemnt this block and use the Python list at the end of this notebook\n",
+    "# if you run into any issues with the translator API\n",
+    "translator= Translator(from_lang=\"autodetect\", to_lang=\"en\")\n",
+    "translations = []\n",
+    "\n",
+    "# Translate text\n",
+    "for text in top_80_df['text']:\n",
+    "    translation = translator.translate(text)\n",
+    "    # If the header is already in english we will just paste the original text\n",
+    "    if translation == 'PLEASE SELECT TWO DISTINCT LANGUAGES':\n",
+    "        translations.append(text) \n",
+    "    else:\n",
+    "        translations.append(translation)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "172418d0-9b7c-4e9f-a390-d0dd5473ec45",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add translated text to the dataframe\n",
+    "top_80_df['translated_text'] = translations\n",
+    "\n",
+    "# Now we can see the translation of that same record\n",
+    "top_80_df['translated_text'].iloc[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bc6f9061-96a4-4d22-af35-8c2f9d9ae19a",
+   "metadata": {},
+   "source": [
+    "### Step 4: Embed and Index Documents"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "41fa6476-9cff-4af0-9d03-50213d72a2e6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Establish Cohere client\n",
+    "co = cohere_aws.Client(mode=cohere_aws.Mode.BEDROCK)\n",
+    "model_id = \"cohere.embed-multilingual-v3\"\n",
+    "\n",
+    "# Embed documents\n",
+    "docs = top_80_df['text'].to_list()\n",
+    "docs_lang = top_80_df['lang'].to_list()\n",
+    "translated_docs = top_80_df['translated_text'].to_list() #for reference when returning non-English results\n",
+    "doc_embs = co.embed(texts=docs, model_id=model_id, input_type='search_document').embeddings\n",
+    "\n",
+    "# Create a search index\n",
+    "index = hnswlib.Index(space='ip', dim=1024)\n",
```

Comment:
Can you add comments for some of these magic numbers?

Reply:
more info added in 67b5f37ca71a4fbfd4a87a9d1671d3319b7b341a